CU-03 Buscar eventos con tu localización y CU-04 Filtrar eventos - Refactorizado y correcciones

angbongon opened the pull request:
Ahora es posible filtrar por categorías (CU-04) además de haberse corregido los errores pertienentes a la búsqueda por geolocalización.

Commit:
[refactor][WIP] Eventos por localización

santiborge97 committed:
[fix] Geolocation with more than 12 objects

Commit:
[WIP][refactor][fix] Categorias en search y corrección nearby

Ahora cuando se hace una búsqueda usando nearby se pueden usar el resto
de filtros manteniendo la localización

Commit:
[WIP][refactor] Se mueven funciones de servicios a selectores

Commit:
[fix] Se elimina de session la longitud

Antes sólo se eliminaba la latitud

Commit:
[fix] Lat y lon se eliminan siempre desde busqueda normal con home

Esto producía que a pesar de buscar con el botón normal se seguía usando
la geolocalización. También producía un error que hacía acceder a las
variables de geolocalización cuando estaban vacías, produciendo un error
500.

Commit:
Merge remote-tracking branch 'remotes/origin/fix/be/Nearby-Search' into refactor/be/CU-03

Commit:
[fix] Filtro por categorías funciona bien con nearby

Commit:
[fix] Se quitan valores fijos de lat y lon

angbongon committed:
Merge branch 'dev' into refactor/be/CU-03

angbongon commented:
@juanmiguelruiz te asigno como reviewer para que puedas mirar como se ve ahora el tema de los botones de categoría.

juanmiguelruiz commented:
Dejalo asi y lo cambiare yo

juanmiguelruiz commented:
acepta la pull

angbongon commented:
> acepta la pull

Espero a que acepte QA y hago el merge

angbongon committed:
Merge branch 'dev' into refactor/be/CU-03

angelcalzadollamas reviewed (changes requested):
**Para corregir:**
Ahora cuando no activas la localización ya no sale el mensaje que puso santi de: No tienes activada la localización. Se debe avisar de ello, aquí una imagen de que no se ve:
![Ya no sale el mensaje de que no has activado la localización](https://user-images.githubusercontent.com/28622424/78982630-d7550800-7b22-11ea-95b3-072febfd0cd0.PNG)

**Proposición de mejora:**
Cuando filtras por una categoría, ¿Por qué se van los cuadros de las demás categorías? ¿No se podrían dejar pa poder ir cambiandolas facilmente? Aquí una imagen de como cuando filtras una categoría desaparecen las demás:
![No podrían seguir saliendo todas las categorías pa seguir cambiando de una a otra](https://user-images.githubusercontent.com/28622424/78982717-01a6c580-7b23-11ea-8b4f-9d1918f96e74.PNG)

juanmiguelruiz commented:
La verdad que este cambio se descartó por ser demasiado complicado por parte del back, ahora lo habéis hecho pero desde el front no estaba contemplado hacer esta vista en la planificación ya que no solo hay que cambiarla en escritorio sino también en todo el responsive. Así que no hay ningún problema en hacerla pero cuando todo lo planificado este hecho.

Si por parte del back no hay que hacer ningún cambio más se deja la pull aquí, hasta que se pueda hacer por parte del front

santiborge97 committed:
[fix]Apaño para mostrar mensaje de activar GPS

santiborge97 commented:
Yo he hecho un commit con un apaño para mostrar el mensaje de error. Míralo a ver si te gusta @angbongon

angelcalzadollamas commented:
Estoy de acuerdo con lo que dice juanmi, recuerdo que comentó que por la parte del front era complicado contemplar lo de las categorías. Sin embargo, lo del mensaje que debe aparecer si el usuario no tiene habilitada la ubicación debe aparecer. Mirare si está en dev y sino pediré que se ponga aunque sea a dev.

_Esta funcionalidad pasa a ser de prioridad baja, es decir, a la espera de que el front termine todas sus cosas y pueda ponerse manos a la obra con esta tarea que se canceló en el sprint 2_

Commit:
[refactor] Usuario no entra en listado si no tiene ubicación

Aparece como mensaje de error en el formulario

Commit:
[hack] Se comenta la parte de categorías en html

angbongon committed:
Merge branch 'dev' into refactor/be/CU-03

Commit:
[refactor] Geolocalización y búsqueda en un mismo form

Para poder solucionar el error de front con el botón de geolocalización
que no se coloca bien

Commit:
[fix] Casting de lat y long a float

juanmiguelruiz committed:
[WIP] Boton localizacion

juanmiguelruiz committed:
Merge branch 'refactor/be/CU-03' of https://github.com/eventshow/eventshow into refactor/be/CU-03

Commit:
[feat] Eventos ordenados por fecha de inicio

juanmiguelruiz committed:
[feat] Button Localizacion

juanmiguelruiz committed:
Merge branch 'refactor/be/CU-03' of https://github.com/eventshow/eventshow into refactor/be/CU-03

juanmiguelruiz committed:
[fix] Resultados busquedas avanzada

angbongon committed:
Merge branch 'dev' into refactor/be/CU-03

juanmiguelruiz committed:
[fix] Form errors home

angbongon committed:
Merge branch 'dev' into refactor/be/CU-03

angelcalzadollamas reviewed (changes requested):
Por qué si yo quiero poner una ubicación que no existe en el autocompletar me sale esto

![Captura](https://user-images.githubusercontent.com/28622424/79049053-85df7280-7c21-11ea-8e38-f9d573348100.PNG)

angbongon commented:
> Por qué si yo quiero poner una ubicación que no existe en el autocompletar me sale esto
> 
> ![Captura](https://user-images.githubusercontent.com/28622424/79049053-85df7280-7c21-11ea-8e38-f9d573348100.PNG)

Porque la suckas. Eso sólo pasa si pulsas intro. La solución es cambiar el orden de los botones en HTML. Si se quiere mantener el orden el cambio se debe hacer en frontend

Commit:
[fix] Orden en HTML de botones submit

Debido al orden de los submits en el HTML al hacer una búsqueda dando a
intro el form se procesa cogiendo el primero de ellos (geolocalización=, a pesar de que no se
esté cogiendo deliberadamente, resultando en que
salte una validación (correctamente).

angelcalzadollamas reviewed (approved):
Todo correcto. Le diré a juanmi que revise una cosa de front cuando esto esté en dev

angbongon commented:
> Todo correcto. Le diré a juanmi que revise una cosa de front cuando esto esté en dev

Calsadito calsadito... Ten cuidaito
